### ISSUE TITLE
Should augmented assignment operator on `TypedDict` fail if type is compatible?

### ISSUE THREAD

**Issue by michaeloliverx** [microsoft/pylance-release#2300]
```python
from typing import TypedDict


class Person(TypedDict, total=False):
    name: str
    age: int


person: Person = {}

# Ok
person.update({"name": "Michael"})

# Expression of type "dict[str, str | object]" cannot be assigned to declared type "Person"
# "dict[str, str | object]" is incompatible with "Person"
person |= {"name": "Michael"}
```

To quote [PEP 584](https://www.python.org/dev/peps/pep-0584/):
> Augmented assignment behaves identically to the ``update`` method called with a single positional argument, so it also accepts anything implementing the Mapping protocol (more specifically, anything with the ``keys`` and ``__getitem__`` methods) or iterables of key-value pairs.

**Comment by erictraut** [comment]
[PEP 589](https://www.python.org/dev/peps/pep-0589/), which introduced `TypedDict`, doesn't specify that the `|` operator is supported by `TypedDict`. It discusses the principles about which operations are allowed in [this section](https://www.python.org/dev/peps/pep-0589/#supported-and-unsupported-operations). These principles do not appear to rule out support for `|` or `|=` as long as both operands are the same `TypedDict` type.

The typeshed type stubs also define a private class called `_TypedDict` that defines (in rough terms) the operations that should be permitted on all `TypedDict` classes. I say "rough terms" because supported methods are generally synthesized by type checkers to provide more specific parameter and return types corresponding to individual fields within the `TypedDict`. Interestingly, the `_TypedDict` class was amended just a few weeks ago by typeshed maintainers to include `__or__` and `__ior__` magic methods, implying that `|` and `|=` should now be allowed.

Pyright (the type checker that underlies pylance) will need to be modified to support this. I've created [this issue](https://github.com/microsoft/pyright/issues/2951) to track that work.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by michaeloliverx** [comment]
> This will be addressed in the next release.

The turnaround on issues here is incredible. Fantastic work 👏 typed Python future is looking bright.

**Comment by debonte** [comment]
This issue has been fixed in version 2022.2.0, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202220-2-February-2022)
